openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -372,8 +372,14 @@ def get_from_id(cls, course_id):
         # Regenerate the thumbnail images if they're missing (either because
         # they were never generated, or because they were flushed out after
         # a change to CourseOverviewImageConfig.
-        if course_overview and not hasattr(course_overview, 'image_set'):
-            CourseOverviewImageSet.create(course_overview)
+        if course_overview:
+            if hasattr(course_overview, 'image_set'):
+                image_set = course_overview.image_set
+                if not image_set.small_url or not image_set.large_url:
+                    CourseOverviewImageSet.objects.filter(course_overview=course_overview).delete()
+                    CourseOverviewImageSet.create(course_overview)
```

Comment:
I'm wondering if you need the delete and create here vs just running an update? Is there a difference in what happens with the ImageSet with that?

Reply:
I'm not positive (bad answer), but I was following how `load_from_module_store()` handles the ImageSet in this file on `line 282`

Reply:
Yeah... We overwrite the create method so trying to do update might not do what we want. This works for me.